src/sentry/uptime/detectors/detector.py

```diff
@@ -0,0 +1,21 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from sentry import features
+from sentry.uptime.detectors.hostname_extraction import extract_hostname_from_url
+from sentry.uptime.detectors.ranking import add_hostname_to_rank
+
+if TYPE_CHECKING:
+    from sentry.models.project import Project
+
+
+def detect_hostname_for_project(project: Project, url: str) -> str | None:
+    if not features.has("organizations:uptime-automatic-hostname-detection", project.organization):
```

Comment:
Flagpole is fast for ingestion stuff now yeah?

Reply:
`project.organization` I assume is already cached

Reply:
Yep, it's all backed by options automator